http_parser.c

```diff
@@ -435,6 +435,17 @@ enum http_host_state
   (IS_ALPHANUM(c) || (c) == '.' || (c) == '-' || (c) == '_')
 #endif
 
+#define NEXTHEADERCHAR()                                      \
+  if (1) {                                                    \
```

Comment:
Using if(1) {} else instead because of the break within this macro

Reply:
I get multiple compiler errors for this same macro, looks like additional braces are needed?:

```
http_parser.c:937:25: error: suggest braces around empty body in an ‘else’ statement [-Werror=empty-body]
         NEXTHEADERCHAR();
```

Reply:
Macros with control flow in them that don't announce it in their name are a special kind of evil.